tests/test_all_integration.py

```diff
@@ -0,0 +1,316 @@
+import tests.context
+import pytest
+import time
+from multiprocessing import Process
+import os
+import shutil
+
+from core.communication_manager         import CommunicationManager
+from core.runner                        import DMLRunner
+from core.scheduler                     import DMLScheduler
+from core.configuration                 import ConfigurationManager
+from tests.testing_utils                import make_initialize_job, make_model_json
+from tests.testing_utils                import make_serialized_job, serialize_job
+from core.utils.enums                   import RawEventTypes, JobTypes, MessageEventTypes
+from core.utils.keras                   import serialize_weights
+from core.blockchain.blockchain_gateway import BlockchainGateway
+from core.blockchain.blockchain_utils   import setter, TxEnum
+
+
+config_manager = ConfigurationManager()
+config_manager.bootstrap(
+    config_filepath='tests/artifacts/integration/configuration.ini'
+)
+
+@pytest.fixture
+def mnist_filepath():
+    return 'tests/artifacts/integration/mnist'
+
+@pytest.fixture(scope='session')
+def transformed_filepath():
+    return 'tests/artifacts/integration/mnist'+'/transformed'
```

Comment:
Why not just put the two strings together?

Reply:
I wanted the first string to not be hardcoded and to be mnist_filepath. But pytest kept yelling at me.

Reply:
More details? Also, why is the scope `session`?

Reply:
Scope is session so that it can be used by the cleanup fixture which has scope=session, and scope is session for the cleanup fixture so that it will run after all the other tests. Fixtures with different scopes can't be used together apparently. So that's why it yells at me for using mnist_filepath.

Reply:
I see. What about making all of the fixtures session scope? IF not, can you put the two strings together?

Reply:
Ah that actually works very well. Note: in future, fixtures should be session scope by default. This speeds up the tests a LOT.

Reply:
Can you refactor relevant fixtures in optimizer/communication_manager tests so that they are session scope too? Too lazy to comment on all of those.

Also nice catch on scope, I actually assumed the default behavior was module the entire time.

Reply:
Done (tests are now 45 seconds faster!)